content/docs/internals/architecture.md

```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
 
 ![pomerium architecture diagram](./img/architecture/pomerium-system-context.svg)
 
-## Component Level
+## Component level
 
-Pomerium is composed of 4 logical components:
+Pomerium is composed of four logical components:
 
-- Proxy Service ([What is a proxy?](https://www.pomerium.com/blog/proxy-vs-reverse-proxy/))
+### Proxy service
+The Proxy service receives requests from the client and routes requests between the client, upstream services, and other Pomerium components. 
 
-  - All user traffic flows through the proxy
-  - Verifies all requests with Authentication service
-  - Directs users to Authentication service to establish session identity
-  - Processes policy to determine external/internal route mappings
-  - Receives user OIDC tokens and session data from Authentication service and stores in Databroker service
+The following steps outline how the Proxy service handles a request at a high level:
 
-- Authentication Service
+1. The client makes a request to access the target resource 
+1. The Proxy service receives the request and redirects it to the Authorization service before proxying it
+1. If the Authorization service doesn’t see a session cookie, the Proxy service redirects the request to the Authentication service to verify the client’s identity
+1. After the Authentication service verifies the client’s identity and saves a local session cookie, the Proxy service redirects the session data to the Databroker service
```

Comment:
it does not redirect the session data to the Databroker service. It makes a gRPC call to the Databroker service to save the session.